tests/test_maps.py

```diff
@@ -0,0 +1,170 @@
+    "map_year" : 3000,
+    "x_coord" : "8",
+    "y_coord" : "9",
+    'links': [],
+    'media': [],
+    'story_ids': []
+}
+
+def find_map_occurence(map_obj):
+    data = requests.get('http://127.0.0.1:5000/maps')
+    map_year = map_obj['map_year']
+    map_year_list = list(filter(lambda x: x['map_year'], data.json()['result']['maps']))
+    while map_year in map_year_list:
+        map_year = random.randInt(1800,2018)
+    map_obj['map_year'] = map_year
+
+# def find_non_existent_id():
+#     _id = 13
+
+
+
+class MapTests(unittest.TestCase):
+ # this is a comprensive test suite for testing the map end points
+
+    def test_map_post_get(self):
+        # testing 1 post and 1 get
+        find_map_occurence(map_add_json_1)
+        post_data = requests.post('http://127.0.0.1:5000/maps', json= map_add_json_1)
+        get_data = requests.get('http://127.0.0.1:5000/maps')
+
+        post_data_body = post_data.json()['result']['map']
+        get_data_body = get_data.json()['result']['maps']
+        self.assertEqual(post_data.json()['code'], 201)
+        self.assertEqual(get_data.json()['code'], 200)
+
+        map_id_list = [x['_id'] for x in get_data_body]
+        map_posted = post_data_body['_id'] in map_id_list
+        self.assertEqual(True, map_posted)
+
+        # is there a way to shorten this ?
+
+        map_in_list = None
+        for map_obj in get_data_body:
+            if map_obj['_id'] == post_data_body['_id']:
+                map_in_list = map_obj
+                break
+
+        self.assertEqual(post_data_body['map_year'], map_in_list['map_year'])
+        self.assertEqual(post_data_body['image_url'], map_in_list['image_url'])
+
+    def test_map_post_get_delete_bad(self):
+        # test that the proper status_code is returned given bad requests
+        before_get = requests.get('http://127.0.0.1:5000/maps')
+        before_size = len(before_get.json()['result']['maps'])
+
+        response = requests.post('http://127.0.0.1:5000/maps', json= map_add_json_bad_1)
+        self.assertEqual(response.json()['code'], 400)
+
+        response = requests.post('http://127.0.0.1:5000/maps', json= map_add_json_bad_2)
+        self.assertEqual(response.json()['code'], 400)
+
+        response = requests.post('http://127.0.0.1:5000/maps', json= map_add_json_bad_3)
+        self.assertEqual(response.json()['code'], 400)
+
+        id = 1
+        while id in list(filter(lambda x: x['_id'], before_get.json()['result']['maps'])):
+            id+=1
+        response = requests.delete('http://127.0.0.1:5000/maps/{}'.format(id))
+        self.assertEqual(response.json()['status'], 404)
+
+        after_get = requests.get('http://127.0.0.1:5000/maps')
+        after_size = len(after_get.json()['result']['maps'])
+        self.assertEqual(before_size,after_size)
+
+        # This test suite is under the assumption that the post_poi/get_poi
+        # is fully functional.
+
+    def test_map_delete(self):
```

Comment:
this should only be testing delete. this is where you should delete the maps you previously created